src/tribler-gui/tribler_gui/core_manager.py

```diff
@@ -83,7 +87,7 @@ def on_core_finished(self, exit_code, exit_status):
                     self.core_traceback_timestamp,
                 )
 
-            raise RuntimeError(exception_msg)
+            raise CoreRuntimeError(exception_msg)
```

Comment:
Can you rename the error to CoreCrashedError to clarify that it does not inherit from RuntimeError?

Reply:
Or even better would be to have base exception class `CoreError` inherited from `RuntimeError` with two subclasses `CoreConnectTimeoutError` and `CoreCrashedError`